app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -60,6 +62,15 @@ class SettingsFragment : PreferenceFragmentCompat(), CoroutineScope, AccountObse
         super.onCreate(savedInstanceState)
         job = Job()
         updateSignInVisibility()
+
+        preferenceManager.sharedPreferences.registerOnSharedPreferenceChangeListener { sharedPreferences, key ->
+            context!!.components.analytics.metrics.track(
+                Event.PreferenceToggled(
```

Comment:
and then here you could "try" to create the event, and then track it if it succeeds.